Bayes Theorem


P(A|B) = P(B|A) P(A) / P(B)

We have 2 buckets A and B. In bucket A we have 30 blue balls and 10 yellow balls, while inn bucket B we have 20 blue and 20 yellow balls. We are required to choose one ball. What is the chance that we choose bucket A?

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
hypos ='Bucket A','Bucket B'

probs= 1/2,1/2

In [ ]:
prior = pd.Series(probs, hypos)
print(prior)

Bucket A    0.5
Bucket B    0.5
dtype: float64


In [ ]:
likehood = 3/4, 1/2
unnorm = prior * likehood
print (unnorm)

Bucket A    0.375
Bucket B    0.250
dtype: float64


In [ ]:
prob_data = unnorm.sum()
prob_data

0.625

In [ ]:
posterior = unnorm / prob_data

In [ ]:
posterior

Bucket A    0.6
Bucket B    0.4
dtype: float64

Here from the result, we can say the posterior probability of choosing bucket A with the blue ball 0.6

Now, what is the probability that both times we chose bucket A to pick the ball ??????

In [ ]:
prior = posterior
print(prior)

Bucket A    0.6
Bucket B    0.4
dtype: float64


In [ ]:
likehood = 3/4, 1/2
unnorm = prior * likehood
print(unnorm)

Bucket A    0.45
Bucket B    0.20
dtype: float64


In [ ]:
prob_data = unnorm.sum()
prob_data


0.6499999999999999

In [ ]:
posterior = unnorm / prob_data
print(posterior)

Bucket A    0.692308
Bucket B    0.307692
dtype: float64


Here we can see that the posterior probability for bucket AA in the second attempt is 0.6923